pkg/api/handlers/libpod/kube.go

```diff
@@ -126,3 +126,29 @@ func KubePlayDown(w http.ResponseWriter, r *http.Request) {
 func KubeGenerate(w http.ResponseWriter, r *http.Request) {
 	GenerateKube(w, r)
 }
+
+func KubeApply(w http.ResponseWriter, r *http.Request) {
+	runtime := r.Context().Value(api.RuntimeKey).(*libpod.Runtime)
+	decoder := r.Context().Value(api.DecoderKey).(*schema.Decoder)
+	query := struct {
+		CACertFile string `schema:"caCertFile"`
```

Comment:
This doesn't look right.  We usually don't allow client to control reading/writing files on the host for security reasons.

Do we need to expose this on the REST API at all?  Couldn't we have `podman` and `podman-remote` call the same function?

We could move `pkg/domain/infra/abi/apply.go` into an internal package `pkg/domain/infra/internal/apply.go` that the `infra/abi` and `infra/tunnel` can call.

Reply:
> Do we need to expose this on the REST API at all? Couldn't we have podman and podman-remote call the same function?

That is a good question. I guess it comes down to: Do we want to API callers use it, e.g. should podman-py or podman-desktop have this functionality?

> We usually don't allow client to control reading/writing files on the host for security reasons.

I don't see how this is an argument? podman-remote run -v allows full access to the file system anyway.

Reply:
> Do we want to API callers use it, e.g. should podman-py or podman-desktop have this functionality?

I don't see a compelling reason.  I see `podman kube apply` more for something during development but would recommend `kubectl` for production use.

> I don't see how this is an argument? podman-remote run -v allows full access to the file system anyway.

That's an unfortunate circumstance but shouldn't justify more unfortunate circumstances.  A probably more convincing reason is that users will expect `podman-remote --file foo` to be evaluated on the _client side_ not on the relative path of the server on a remote machine.

Reply:
> > I don't see how this is an argument? podman-remote run -v allows full access to the file system anyway.
>
> That's an unfortunate circumstance but shouldn't justify more unfortunate circumstances. A probably more convincing reason is that users will expect podman-remote --file foo to be evaluated on the client side not on the relative path of the server on a remote machine.

Sure I agree there and it would make it more consistent with the other commands. Just saying we should not argue with security. The train is long gone, access to the podman socket allows arbitrary code execution, this is by design of how podman run works.

Reply:
You are correct to put it into context but I will always argue with security.  One potentially dangerous feature should not justify making others less secure.  It's possible to deny the use of "dangerous" parameters, for instance.  The more there are, the harder it gets.